src/core/methods/createChannel.ts

```diff
@@ -3,134 +3,74 @@
 import { ad4mClient } from "@/app";
 import { CHANNEL } from "@/constants/neighbourhoodMeta";
+import { useDataStore } from "@/store/data";
+import { useAppStore } from "@/store/app";
 
 interface ChannelProps {
   channelName: string;
   creatorDid: string;
   sourcePerspective: PerspectiveHandle;
-  membraneType: MembraneType;
-  typedExpressionLanguages: FluxExpressionReference[];
 }
 
 export async function createChannel({
   channelName,
   creatorDid,
   sourcePerspective,
-  membraneType,
-  typedExpressionLanguages,
 }: ChannelProps): Promise<ChannelState> {
-  const perspective = await ad4mClient.perspective.add(channelName);
-  console.debug("Created new perspective with result", perspective);
-  const socialContextLanguage =
-    await ad4mClient.languages.applyTemplateAndPublish(
-      SOCIAL_CONTEXT_OFFICIAL,
-      JSON.stringify({
-        uid: v4().toString(),
-        name: `${channelName}-social-context`,
-      })
-    );
-  console.debug(
-    "Created new social context language wuth result",
-    socialContextLanguage
-  );
-
-  //Publish perspective
-  const metaLinks = await createNeighbourhoodMeta(
-    channelName,
-    "",
-    creatorDid,
-    typedExpressionLanguages
-  );
-
-  const meta = new Perspective(metaLinks);
+  const dataStore = useDataStore();
+  const appStore = useAppStore();
 
-  const neighbourhood = await ad4mClient.neighbourhood.publishFromPerspective(
-    perspective.uuid,
-    socialContextLanguage.address,
-    meta
-  );
-  console.debug("Create a neighbourhood with result", neighbourhood);
+  const channel = dataStore.channels[channelName];
+  if (!channel || channel.sourcePerspective !== sourcePerspective.uuid) {
+    const addLinkToChannel = await ad4mClient.perspective.addLink(
+      sourcePerspective.uuid,
+      {
+        source: sourcePerspective.sharedUrl!,
+        target: channelName,
+        predicate: CHANNEL,
+      }
+    );
 
-  const addLinkToChannel = await ad4mClient.perspective.addLink(
-    sourcePerspective.uuid,
-    {
-      source: sourcePerspective.sharedUrl!,
-      target: neighbourhood,
-      predicate: CHANNEL,
-    }
-  );
-  console.debug(
-    "Created new link on source social-context with result",
-    addLinkToChannel
-  );
+    console.debug(
+      "Created new link on source social-context with result",
+      addLinkToChannel
+    );
 
-  //Add link on channel social context declaring type
-  const addChannelTypeLink = await ad4mClient.perspective.addLink(
-    perspective.uuid,
-    {
-      source: neighbourhood,
-      target: CHANNEL,
-      predicate: "rdf://type",
-    }
-  );
-  console.log(
-    "Added link on channel social-context with result",
-    addChannelTypeLink
-  );
+    const channelCreatorDidLink = await ad4mClient.perspective.addLink(
+      sourcePerspective.uuid,
+      {
+        source: addLinkToChannel.data.target,
+        target: creatorDid,
+        predicate: 'flux://creator_did',
```

Comment:
Better make it a constant, also should be verb for predicate, like `created_by`.